components/financial_assistance/app/domain/financial_assistance/operations/applications/verifications/validate_applicant_and_update_evidence.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+module FinancialAssistance
+  module Operations
+    module Applications
+      module Verifications
+        # This class handles the validation of an applicant and updates the evidences as needed
+        class ValidateApplicantAndUpdateEvidence
+          include Dry::Monads[:result, :do]
+
+          EVIDENCE_ALIASES = {
+            income: :income_evidence,
+            esi_mec: :esi_evidence,
+            non_esi_mec: :non_esi_evidence,
+            local_mec: :local_mec_evidence
+          }.freeze
+
+          def call(payload_entity, application)
+            return handle_malformed_payload_entity(payload_entity, application) if payload_entity.failure?
```

Comment:
@TristanB17 Please handle the failure in the Parent Class "RequestEvidenceDetermination" and also add the type history elements to the evidences.

Reply:
@saikumar9 And adding those type_history elements should only the added in the case of a malformed FA application, correct?

Reply:
@TristanB17 I am suggesting you move the logic into Parent Class with whatever you are doing if there are errors(adding this information to the type history elements of the evidences).

Reply:
@saikumar9 Moved the handle_malformed_payload_entity method into the parent class, should be good to go!